### PR TITLE
Symintified factory functions

### PR DESCRIPTION
**Pull request by Chillee**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #86067
* #86104

### CODE REVIEW

File: torch/fx/experimental/proxy_tensor.py [hunk 1]
```diff
@@ -384,6 +384,7 @@ def create_arg(self, a: Any):
 
             return self.create_node('get_attr', qualname, (), {})
         elif isinstance(a, (SymInt, SymFloat)):
+            breakpoint()
```
Comment (ezyang) [inline torch/fx/experimental/proxy_tensor.py:387]: Remove

**Review by ezyang — APPROVED** [review]

File: torch/fx/experimental/proxy_tensor.py (hunk 1 above)
Comment (Chillee) [reply torch/fx/experimental/proxy_tensor.py:387]: There's a weird failure here I was debugging. For some reason the assert triggers 🤔

**Comment by Chillee** [comment]
We have some .... wrong behavior when passing in lists of mixed tensors/ints to ops that expect symints. Seem like we just drop the tensors.

```
def f(tensors):
    max_size = torch.tensor([800, 1216], dtype=torch.int64)
    batch_shape = [len(tensors)] + list(tensors[0].shape[:-2]) + list(max_size)
    print(batch_shape)
    return tensors[0].new_empty(batch_shape)

a = torch.randn(3, 800, 1199)
b = torch.randn(3, 800, 800)
print(make_fx(f, tracing_mode="symbolic")([a, b]).code)
```
cc: @ezyang @eellison

**Comment by ezyang** [comment]
In IntList we have

```
      // convert tensor to scalar outside of try / catch,
      // so that Tensor subclass exceptions will not be caught.
      if (THPVariable_Check(obj)) {
        auto& var = THPVariable_Unpack(obj);
        if (var.numel() != 1 ||
            !at::isIntegralType(
                var.dtype().toScalarType(), /*include_bool*/ true)) {
          throw_intlist_exception(this, i, obj, idx);
        }
        res[idx] = var.item<int64_t>();
      } else {
        try {
          res[idx] = THPUtils_unpackIndex(obj);
        } catch (std::exception& e) {
          throw_intlist_exception(this, i, obj, idx);
        }
```

but the corresponding logic is missing in symintlist in torch/csrc/utils/python_arg_parser.h

**Comment by ezyang** [comment]
https://github.com/pytorch/pytorch/pull/86126

**Pull request closed without merging**